src/script/components/MessagesList/Message/ContentMessage/asset/FileAssetComponent.tsx

```diff
@@ -94,14 +97,14 @@ const FileAsset: React.FC<FileAssetProps> = ({
           role="button"
           tabIndex={messageFocusedTabIndex}
           aria-label={`${t('conversationContextMenuDownload')} ${fileName}.${fileExtension}`}
-          onClick={() => {
+          onClick={async () => {
             if (isUploaded) {
-              downloadAsset(asset);
+              await onDownloadAsset();
```

Comment:
What about moving this if statement to the `onDownloadAsset` function? We could just pass the function name to `onClick`